utils/connects.stanza

```diff
@@ -12,41 +12,90 @@ defpackage ocdb/connects :
   import ocdb/bundles
   import ocdb/symbols
   import ocdb/box-symbol
+  import lang-utils
 
-public defn connect-isolate (a:JITXObject, b:JITXObject) :
-  inside pcb-module :
-    eval-when true :
-      match(port-type(a), port-type(b)) :
-        (pa:Bundle, pb:Bundle) :
-          if pa == i2c and pb == i2c :
-            inst buf : ocdb/texas-instruments/ISO1540/component
-            net (a.sda, buf.SDA1)
-            net (a.scl, buf.SCL1)
-            net (b.sda, buf.SDA2)
-            net (b.scl, buf.SCL2)
-            property(buf.VCC1.requires-power) = true
-            property(buf.VCC1.power-request) = [3.3 0.3 0.05]
+; Helper function to check if the ports have bundles and their assigned
+; bundle matches what we expect
+defn ensure-bundles!<?T> (
```

Comment:
This function is too scary for JITX users. OCDB needs to be written with the perspective that JITX users will scan through it and write more code in the same style. Change the design of this function to be more straightforward.

The current design allows you to write this (at the cost of a very complicated helper function):

```
within ensure-bundles!([x, y], [i2c, i2c], "call connect-isolation"):
   inside pcb-module:
     public inst buf: buffer
     net (x, buf.p[1])
     net (y, buf.p[2])
     buf
```       

Change the design to allow for this style instead.

```
inside pcb-module :
  ensure-bundle-port!(x, i2c)
  ensure-bundle-port!(y, i2c)
  public inst buf: buffer
  net (x, buf.p[1])
  net (y, buf.p[2])
  buf
```

Reply:
The reason for the use of this helper function was to check that all bundles matched their expected bundles and report them altogether, as opposed to checking one at a time and failing at the first change.

With the change to `ensure-bundle-port!` I don't see a clean way of representing this - would `error-buffer` be appropriate, or is this a pattern to avoid in general?

Reply:
I see. You can do this also:
```
ensure-bundle-port!([x, y], [i2c, i2c])
```
The purpose is to reduce the complexity of the `ensure-bundles!` helper.